src/rgw/rgw_sal_motr.cc

```diff
@@ -3422,6 +3583,50 @@ int MotrStore::init_metadata_cache(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+int MotrStore::fdmi_service_start(struct m0_client *m0c)
+{
+  struct m0_reqh *reqh = &m0c->m0c_reqh;
+  struct m0_reqh_service_type *stype;
+  bool start_service = false;
+  int rc = 0;
+
+  stype = m0_reqh_service_type_find("M0_CST_FDMI");
+  ldout(cctx, 0) << "m0_reqh_service_type_find() "  << dendl;
+  if (stype == NULL) {
+    return-EINVAL;
+   }
+
+   fdmi_service = m0_reqh_service_find(stype, reqh);
+   if (fdmi_service == NULL) {
+     ldout(cctx, 0) << "m0_reqh_service_find(): fdmi_service == NULL, allocate "  << dendl;
+     rc = m0_reqh_service_allocate(&fdmi_service, &m0_fdmi_service_type, NULL);
+     if (rc != 0)
+       return rc;
+     ldout(cctx, 0) << "m0_reqh_service_init() "  << dendl;
+     m0_reqh_service_init(fdmi_service, reqh, NULL);
+     start_service = true;
+    }
+
+    if (start_service) {
+      rc = m0_reqh_service_start(fdmi_service);
+      ldout(cctx, 0) << "m0_reqh_service_start(): rc =  " << rc << dendl;
+    }
+    return rc;
+}
+
+void MotrStore::fdmi_service_stop(struct m0_client *m0c)
```

Comment:
Where is it called?

Reply:
it should be called the same place as m0_client_fini(), as mgw doesn't finalise motr instance (I don't remember we do it, I think we missed it).

Reply:
I think we call now m0_client_fini() now. So let's call fdmi_service_stop() also beside it?

Reply:
yes, agree